bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/timingwindowcheck/TimingWindowCheckCron.scala

```diff
@@ -15,14 +20,56 @@ object TimingWindowCheckCron {
       Action.delay(logger.log(level, msg))
   }
 
+  private def sendEmails(
+    log: ActionLogger,
+    now: Instant,
+    odb: IDBDatabaseService,
+    twcm: TimingWindowCheckMailer,
+    ps:  TimingWindows
+  ): Action[Unit] =
+
+    ps.toList.traverseU {
+      case (pid, l) =>
+        l.toNel.fold(Action.unit) { tws =>
+          EitherT(ProgramAddresses.fromProgramId(odb, pid).catchLeft).flatMap {
+            case None =>
+              log.log(Level.INFO, s"Could not get email addresses for $pid because it was not found in ODB")
+
+            case Some(Failure(msg)) =>
+              log.log(Level.INFO, s"Could not send mask check nag email because some addresses are not valid for $pid: $msg")
+
+            case Some(Success(pa)) =>
+              twcm.notifyPendingCheck(pid, pa, tws)
+          }
+        }
+    }.void
+
   /** Cron job entry point.  See edu.gemini.spdb.cron.osgi.Activator. */
   def run(ctx: BundleContext)(tmpDir: File, logger: Logger, env: java.util.Map[String, String], user: java.util.Set[Principal]): Unit = {
 
+    def getCheckUnion(now: Instant, site: Site): Union[Interval] = {
+
+      val night = TwilightBoundedNight.forInstant(TwilightBoundType.NAUTICAL, now, site)
```

Comment:
I think if this is called during the night, it will get you the night you want.  On the other hand if called during the day, it will get you the coming night.